### PR TITLE
Add support for directory content copy [specific ci=1-44-Docker-CP-Online]

### LINKED ISSUE #5796: Support docker cp to copy content of a directory using /.
As a user, I expect the docker cp command to copy the content of `dir` by specifying src path as `/path/to/dir/.` 

Expect behavior:
`docker cp test/. cid:/tmp`  and `docker cp cid:/tmp/. newdir/` 
should work for both offline and online containers

### PR DESCRIPTION
**Pull request by morris-jason**
Fixes #5796

### CODE REVIEW

File: tests/test-cases/Group1-Docker-Commands/1-44-Docker-CP-Online.robot [hunk 1]
```diff
@@ -164,7 +162,7 @@ Copy a large file to an online container, dst is a volume
 
 Copy a non-existent file out of an online container
     ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} cp online:/dne ${CURDIR}
-    Should Not Be Equal As Integers  ${rc}  0
+    Should Be Equal As Integers  ${rc}  0
     Should Contain  ${output}  Error
```
Comment (jialin-li) [inline tests/test-cases/Group1-Docker-Commands/1-44-Docker-CP-Online.robot]: Should change it to `Should Not Contain  ${output}  Error`

File: lib/archive/util.go [hunk 2]
```diff
@@ -106,20 +115,26 @@ func AddMountInclusionsExclusions(currentMount string, filter *FilterSpec, mount
...
 
-// removeSlashes will remove a leading and trailing '/' from
-// the target path
-func removeSlashes(path string) string {
-	return strings.TrimPrefix(strings.TrimSuffix(path, "/"), "/")
+	if leading {
+		path = strings.TrimPrefix(path, "/")
+	}
+	if trailing {
+		path = strings.TrimSuffix(path, "/")
```
Comment (jialin-li) [inline lib/archive/util.go]: filePath.Clean() strips trailing `/` already, no need to trim again.

File: tests/test-cases/Group1-Docker-Commands/1-44-Docker-CP-Online.robot (hunk 1 above)
Comment (morris-jason) [inline tests/test-cases/Group1-Docker-Commands/1-44-Docker-CP-Online.robot]: 👍

File: lib/archive/util.go (hunk 2 above)
Comment (morris-jason) [inline lib/archive/util.go]: Oh cool :)

File: lib/archive/util.go [hunk 3]
```diff
@@ -51,20 +51,29 @@ func GenerateFilterSpec(copyPath string, mountPoint string, primaryTarget bool,
...
-
-	//we only need the right most element.
-	_, first := filepath.Split(copyPath)
+	var first string
+
+	// If the copyPath ends with '/.', we don't want the content's directory name
+	// to be included in the resulting tar header. we do this by removing the directory
+	// portion of the rebase path.
+	// if the copyPath does not end with '/.', we only need the right most element of copyPath.
+	if !strings.HasSuffix(copyPath, "/.") {
```
Comment (hickeng) [inline lib/archive/util.go]: pedantically this should be `filepath.Base(copyPath) == "."`  and I'm not sure why you're then calling `filepath.Split` on it afterwards as `first` will always be `"."` given the input.

Comment (morris-jason) [inline lib/archive/util.go]: Hmm, was just modifying current code, but it seems like you're right. I'll refactor and retest.

File: lib/archive/util.go [hunk 4]
```diff
@@ -106,25 +120,26 @@ func AddMountInclusionsExclusions(currentMount string, filter *FilterSpec, mount
 	for _, mount := range mounts {
 		if strings.HasPrefix(mount, currentMount) && currentMount != mount {
 			// exclusions are relative to the mount so the leading `/` should be removed unless we decide otherwise.
-			exclusion := removeLeadingSlash(strings.TrimPrefix(mount, currentMount))
-			if !strings.HasSuffix(exclusion, "/") {
-				exclusion = exclusion + "/"
-			}
-
+			exclusion := clean(strings.TrimPrefix(mount, currentMount), true)
+			exclusion = strings.TrimPrefix(exclusion, "/") + "/"
```
Comment (hickeng) [inline lib/archive/util.go]: don't know that this second TrimPrefix is needed given the call to `clean(..., true)`

Comment (morris-jason) [inline lib/archive/util.go]: Yeah, it should just be exclusion = clean(....) + '/',  without the second line.

**Comment by morris-jason** [comment]
Closing in favor of #5875 5875

**Pull request closed without merging**